### PR TITLE
[189] Allow jobseekers to hide their profile from organisations

### PR DESCRIPTION
**Pull request by elliotcm**
## Changes in this PR:

See the "Hide profile from schools or trusts" section of the jobseeker profile.

### CODE REVIEW

**Comment by elliotcm** [comment]
[Candidates can hide their profile from organisations](https://trello.com/c/w4mB4hzQ/189-candidates-can-hide-their-profile-from-organisations)

**Comment by bencmitchell** [comment]
@elliotcm this all looks good to me 👍

File: app/controllers/jobseekers/profiles/hide_profile_controller.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    @form = Jobseekers::Profile::HideProfileForm.new(hide_profile: profile.hide_profile)
+  end
+
+  def confirm_hide
+    form_params = params
+      .require(:jobseekers_profile_hide_profile_form)
+      .permit(:hide_profile)
+
+    if (@form = Jobseekers::Profile::HideProfileForm.new(form_params)).valid?
+      profile.update(hide_profile: @form.hide_profile)
```
Comment (BroiSatse) [inline app/controllers/jobseekers/profiles/hide_profile_controller.rb:12]: Nitpick: should we use `update!` here instead - non-banged version risks failed save without any feedback

**Comment by BroiSatse** [comment]
I found one interesting scenario here - if a user selects "yes" on the first question, and then cancels it results in profile displaying "Yes" next to hide from organisations with no organisation listed. The same happens when the last organisation is removed from the list. It's a minor thing, not even sure if it is an issue, but I'd imagine it should change to "No" if the list is empty.

**Review by BroiSatse — APPROVED** [review]

**Comment by elliotcm** [comment]
> I found one interesting scenario here - if a user selects "yes" on the first question, and then cancels it results in profile displaying "Yes" next to hide from organisations with no organisation listed. The same happens when the last organisation is removed from the list. It's a minor thing, not even sure if it is an issue, but I'd imagine it should change to "No" if the list is empty.

@BroiSatse Yeah the behaviour of that page when there's no orgs set (it's the same if you delete them all) is a bit weird but that's how the prototype works.

**Comment by elliotcm** [comment]
> The only question I have is if you checked `i18n` phrases with someone from content. Other than that super solid! 🚀

@ddippolito All the copy came straight from the prototype.

**Pull request merged**